### ISSUE TITLE
There should be a way to update the deep link in an already open banner

### ISSUE THREAD

**Issue by stefkoenig** [BranchMetrics/web-branch-deep-linking#423]
Hi Branch Team,

Dollar Shave Club is working on integrating your web SDK. We have a single page app and need a way to dynamically update the link in an already open banner as we switch routes without calling a journeys event again. Let me know if this is a feature you are working on or would consider adding.

Thanks so much for your help!

Stef

**Comment by rubinsingh** [comment]
Hi Stef,

We have a similar feature that is about to be released. What you’ll be able to do is programmatically close and then show a Journey. Before the Journey is shown, you'll be able to change the link data behind the CTA button through a call to branch.setBranchViewData(linkData). 

Thanks,

Rubin

**Comment by stefkoenig** [comment]
Hi Rubin,

Thanks for your fast response! It's nice to hear you are looking into these things. 

We implemented something similar to the workflow you are describing locally, but the user experience of having the banner close and reopen on every page seemed a bit messy. Additionally, if the Journey event was called again too soon, black space was appended to the DOM between the Branch banner and our header. The third issue we ran into was that sometimes our state was incorrectly being tracked and based on the event listeners built into Branch, a `didCloseBanner` event was fired, but the banner wasn't actually closed. Some of these issues may be specific to our use case (SPA on Ember).

Because of this, we were hoping for an experience that would change the link without having to close and reopen the banner. Since that doesn't seem to be coming down the pipeline, I think we may end up using a custom solution. Thanks for your help!

Stef
